luigi/util.py

```diff
@@ -281,30 +281,59 @@ def requires(self):
             def run(self):
                print self.n # this will be defined
                # ...
+
+    If you want to override specific parameters of the base class,
+    you can add them in your call to the context manager:
+
+    .. code-block:: python
+
+        class TaskA(luigi.Task):
+            param_a1 = luigi.IntParameter()
+            param_a2 = luigi.IntParameter()
+
+
+        @inherits(TaskA, param_a1=3)
+        class TaskB(luigi.Task):
+            param_b = luigi.IntParameter()
+
+            # TaskB will not have param_a1 defined
+
+            def requires(self):
+                return self.clone_parent()
+                # but TaskA will have param_a2 already set.
+
+    If more than one task is given in the :class:`~luigi.util.inherits` call, the default
+    arguments are used for all of them.
+
     """
 
-    def __init__(self, *tasks_to_inherit):
+    def __init__(self, *tasks_to_inherit, **non_copy_defaults):
         super(inherits, self).__init__()
         if not tasks_to_inherit:
             raise TypeError("tasks_to_inherit cannot be empty")
 
         self.tasks_to_inherit = tasks_to_inherit
 
+        # If specified by the user, override these parameters by default in the base class
+        self._non_copy_defaults = non_copy_defaults
+
     def __call__(self, task_that_inherits):
         # Get all parameter objects from each of the underlying tasks
         for task_to_inherit in self.tasks_to_inherit:
             for param_name, param_obj in task_to_inherit.get_params():
                 # Check if the parameter exists in the inheriting task
-                if not hasattr(task_that_inherits, param_name):
+                if not hasattr(task_that_inherits, param_name) and param_name not in self._non_copy_defaults.keys():
                     # If not, add it to the inheriting task
                     setattr(task_that_inherits, param_name, param_obj)
 
         # Modify task_that_inherits by adding methods
         def clone_parent(_self, **kwargs):
+            kwargs.update(**self._non_copy_defaults)
```

Comment:
Can this ever be passed by reference? If so do you modify the input?

Reply:
The documentation says:

> If the form “**identifier” is present, it is initialized to a new ordered mapping receiving any excess keyword arguments, defaulting to a new empty mapping of the same type

(taken from https://docs.python.org/3/reference/compound_stmts.html#function-definitions), so I guess this is ok.

Reply:
Thanks for checking!